scene/2d/physics_body_2d.cpp

```diff
@@ -1066,8 +1067,11 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 	Vector2 body_velocity = p_linear_velocity;
 	Vector2 body_velocity_normal = body_velocity.normalized();
 	Vector2 up_direction = p_up_direction.normalized();
-
+	// Hack in order to work with calling from _process as well as from _physics_process; calling from thread is risky
+	float delta = Engine::get_singleton()->is_in_physics_frame() ? get_physics_process_delta_time() : get_process_delta_time();
+	colliders.clear();
 	Vector2 current_floor_velocity = floor_velocity;
+
 	if (on_floor && on_floor_body.is_valid()) {
```

Comment:
It looks like the lag issue on walls is because this block is called only for floor and not wall. Since this PR now applies moving platform movements on wall connections, this condition should be `if ((on_floor || on_wall) && on_floor_body.is_valid())`